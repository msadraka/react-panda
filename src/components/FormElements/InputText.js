// @flow
import React, { PureComponent } from 'react';
import { Input } from 'antd';
import type { FormProps } from 'redux-form';

import StyleWrapper from './input.style';

type Props = {
  placeholder: string,
  disabled: boolean,
  className: string,
  label: string,
  search: boolean,
  tabIndex: string,
  type: string
} & FormProps;

export default class InputText extends PureComponent<Props> {
  render() {
    const {
      input,
      label,
      placeholder,
      className = '',
      meta: { touched, error },
      type,
      search = false,
      disabled = false,
      ...rest
    } = this.props;

    const InputComponent = search ? Input.Search : Input;

    return (
      <StyleWrapper>
        <div
          className={`c-input text ${className} ${
            touched && error ? 'hasError' : ''
          }`}
        >
          <label
            className={`${disabled ? 'disabled' : ''}`}
            htmlFor={input.name}
          >
            {label && <span className="labelText">{label}</span>}

            <InputComponent
              {...input}
              {...rest}
              autoComplete="off"
              id={input.name}
              className={`roundInput${disabled ? ' not-allowed' : ''}`}
              disabled={disabled}
              placeholder={placeholder}
              type={type}
            />
          </label>
          {touched && error && (
            <div className="text-danger">
              <span className="text-danger--text">{error}</span>
            </div>
          )}
        </div>
      </StyleWrapper>
    );
  }
}
