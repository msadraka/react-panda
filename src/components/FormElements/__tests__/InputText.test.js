import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import InputText from '../InputText';

global.Input = require('antd').Input;

describe('<InputText />', () => {
  const tree = props =>
    renderer
      .create(
        <MemoryRouter>
          <InputText
            {...props}
            input={{ name: 'field' }}
            meta={{ touched: false, error: null }}
          />
        </MemoryRouter>
      )
      .toJSON();

  it('renders  without error', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: null }
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders error, not touched', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: 'Required' }
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders error, and touched', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: true, error: 'Required' }
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders search', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: null },
      search: true
    };
    expect(tree(props)).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const props = {
      input: { name: 'field' },
      label: 'field',
      meta: { touched: false, error: null },
      disabled: true
    };
    expect(tree(props)).toMatchSnapshot();
  });
});
