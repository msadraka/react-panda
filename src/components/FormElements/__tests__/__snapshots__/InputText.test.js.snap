// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> renders  without error 1`] = `
<div
  className="c-input text  "
>
  <label
    className=""
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <input
      autoComplete="off"
      className="ant-input roundInput"
      disabled={false}
      id="field"
      name="field"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<InputText /> renders disabled 1`] = `
<div
  className="c-input text  "
>
  <label
    className="disabled"
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <input
      autoComplete="off"
      className="ant-input ant-input-disabled roundInput not-allowed"
      disabled={true}
      id="field"
      name="field"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<InputText /> renders error, and touched 1`] = `
<div
  className="c-input text  "
>
  <label
    className=""
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <input
      autoComplete="off"
      className="ant-input roundInput"
      disabled={false}
      id="field"
      name="field"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<InputText /> renders error, not touched 1`] = `
<div
  className="c-input text  "
>
  <label
    className=""
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <input
      autoComplete="off"
      className="ant-input roundInput"
      disabled={false}
      id="field"
      name="field"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<InputText /> renders search 1`] = `
<div
  className="c-input text  "
>
  <label
    className=""
    htmlFor="field"
  >
    <span
      className="labelText"
    >
      field
    </span>
    <span
      className="ant-input-search roundInput ant-input-affix-wrapper"
    >
      <input
        autoComplete="off"
        className="ant-input"
        disabled={false}
        id="field"
        name="field"
        onChange={[Function]}
        onKeyDown={[Function]}
        style={null}
        type="text"
        value=""
      />
      <span
        className="ant-input-suffix"
      >
        <i
          aria-label="icon: search"
          className="anticon anticon-search ant-input-search-icon"
          onClick={[Function]}
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="search"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </i>
      </span>
    </span>
  </label>
</div>
`;
